backends/advanced/src/advanced_omi_backend/clients/audio_stream_client.py

```
@@ -222,7 +226,22 @@ async def send_audio_stop(self) -> None:
+        logger.info(
+            f"Sent audio-stop (total: {self.chunk_count} chunks, {self.total_bytes} bytes)"
+        )
+
+    async def send_button_event(self, button_state: str = "SINGLE_PRESS") -> None:
+        """Send button event via Wyoming protocol.
+
+        Args:
+            button_state: Button state ("SINGLE_PRESS" or "DOUBLE_PRESS")
+        """
+        if not self.ws:
+            raise RuntimeError("Not connected. Call connect() first.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)